src/SDL12_compat.c

```diff
@@ -5677,7 +5678,7 @@ SDL_ListModes(const SDL12_PixelFormat *format12, Uint32 flags)
         return (SDL12_Rect **) -1;  /* 1.2's dummy driver always returns -1, and it's useful to special-case that. */
     }
 
-    if (!(flags & SDL12_FULLSCREEN)) {
+    if (!windowed_mode_list && !(flags & SDL12_FULLSCREEN)) {
```

Comment:
```suggestion
    if (!(flags & SDL12_FULLSCREEN) && !windowed_mode_list) {
```
Very minor tweak, we only check windowed_mode_list when we know that we're looking for windowed modes.